core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -373,6 +376,41 @@ object DumpLogSegments {
     }
   }
 
+  private class MetadataLogMessageParser extends MessageParser[String, String] {
+    override def parse(record: Record): (Option[String], Option[String]) = {
+      // Per KIP-631, the value format is: int16 (api key), int16 (api version), bytes (encoded ApiMessage)
```

Comment:
I think the kip originally specified int16 for both these fields, but the latest version indicates unsigned ints. Not sure if any of the current branch is depending on either format.

Reply:
Good catch. I'll fix this